From ed727ba37f13de73eb630c145ef87416378d0b1e Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sun, 3 Jun 2018 08:24:56 +0100
Subject: [add] hackerrank, 'migratory birds', python solution

---
 .../migratory.birds.py                             | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.migratory.birds/migratory.birds.py b/hackerrank/algorithms.-.implementation.-.easy.-.migratory.birds/migratory.birds.py
index ccb1532..90209f9 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.migratory.birds/migratory.birds.py
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.migratory.birds/migratory.birds.py
@@ -8,6 +8,20 @@ import sys
 
 # Complete the migratoryBirds function below.
 def migratoryBirds(ar):
+    counts = {}
+    for idx in ar:
+        if not idx in counts:
+            counts[idx] = 0
+        counts[idx] += 1
+    max = 0
+    max_id = None;
+    for id, count in counts.items():
+        if count > max:
+            max = count
+            max_id = id
+        elif count == max and (not max_id or id < max_id):
+            max_id = id
+    return max_id
 
 if __name__ == '__main__':
     fptr = open(os.environ['OUTPUT_PATH'], 'w')
-- 
2.23.0

