From d0b8c80f03a2e42a1b50a6e55ccbaf7c4c247fcc Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 6 Jun 2018 08:48:19 +0100
Subject: [add] hackerrank, 'the hurdle race', c# port

---
 .../the.hurdle.race.cs                                           | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.the.hurdle.race/the.hurdle.race.cs b/hackerrank/algorithms.-.implementation.-.easy.-.the.hurdle.race/the.hurdle.race.cs
index 5524211..4956627 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.the.hurdle.race/the.hurdle.race.cs
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.the.hurdle.race/the.hurdle.race.cs
@@ -17,6 +17,7 @@ class Solution {
 
     // Complete the hurdleRace function below.
     static int hurdleRace(int k, int[] height) {
+        return Max(height.OrderBy(x => x).Last() - k, 0);
     }
 
     static void Main(string[] args) {
-- 
2.23.0

