From 338077e20c028f90fb651421862dda62e6816c5e Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Thu, 28 Nov 2019 22:39:00 +0000
Subject: [add] hackerrank, 'fair rations', go port

---
 .../fair.rations.go                           | 35 +++++++++++++++++--
 1 file changed, 32 insertions(+), 3 deletions(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.go b/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.go
index 72f32ea..f899fb5 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.go
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.go
@@ -10,9 +10,38 @@ import (
 )
 
 // Complete the fairRations function below.
-func fairRations(B []int32) int32 {
-
+func fairRations(B []int32) string {
+    /*
+    1 2 3
+    2 3 3
+    2 4 4
+
+    1 3 1
+    2 4 1
+
+    reject odd number of odds
+    */
+
+    l_odds := 0
+    for _, x := range(B) {
+        if x % 2 == 1 {
+            l_odds += 1
+        }
+    }
+    if l_odds % 2 == 1 {
+        return "NO"
+    }
 
+    mod := 0
+    l_B := len(B)
+    for l := 0; l < l_B - 1; l += 1 {
+        if int(B[l]) % 2 == 1 {
+            B[l] += 1
+            B[l + 1] += 1
+            mod += 2
+        }
+    }
+    return strconv.FormatInt(int64(mod), 10)
 }
 
 func main() {
@@ -42,7 +71,7 @@ func main() {
 
     result := fairRations(B)
 
-    fmt.Fprintf(writer, "%d\n", result)
+    fmt.Fprintf(writer, "%v\n", result)
 
     writer.Flush()
 }
-- 
2.23.0

