From 39be631736e46572a79203d0bca145e2762e6c6c Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Tue, 29 May 2018 22:25:42 +0100
Subject: [add] hackerrank, 'strong password', python port

---
 .../strong.password.py                           | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/hackerrank/algorithms.-.strings.-.easy.-.strong.password/strong.password.py b/hackerrank/algorithms.-.strings.-.easy.-.strong.password/strong.password.py
index cde68bf..0490ec0 100644
--- a/hackerrank/algorithms.-.strings.-.easy.-.strong.password/strong.password.py
+++ b/hackerrank/algorithms.-.strings.-.easy.-.strong.password/strong.password.py
@@ -9,6 +9,22 @@ import sys
 # Complete the minimumNumber function below.
 def minimumNumber(n, p):
     # Return the minimum number of characters to make the password strong
+    special_characters = "!@#$%^&*()+-";  # position of dash matters!
+
+    # initially
+    to_add = 0;
+    # constraints
+    if not re.match("[0-9]", p):
+        to_add += 1;
+    if not re.match("[a-z]", p):
+        to_add += 1;
+    if not re.match("[A-Z]", p):
+        to_add += 1;
+    if not re.match("[" + re.escape(special_characters) + "]", p):
+        to_add += 1;
+    if len(p) + to_add < 6:
+        to_add += (6 - (len(p) + to_add));
+    return to_add;
 
 
 if __name__ == '__main__':
-- 
2.23.0

