From 5bcb8438ca84960fbce4d972ca3f8652ab5c37c1 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Fri, 20 Jul 2018 11:04:24 +0100
Subject: [add] hackerrank, 'mean var and std', python solution

---
 .../mean.var.and.std.py                               | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hackerrank/python.-.numpy.-.easy.-.mean.var.and.std/mean.var.and.std.py b/hackerrank/python.-.numpy.-.easy.-.mean.var.and.std/mean.var.and.std.py
index c2e5936..ed51288 100644
--- a/hackerrank/python.-.numpy.-.easy.-.mean.var.and.std/mean.var.and.std.py
+++ b/hackerrank/python.-.numpy.-.easy.-.mean.var.and.std/mean.var.and.std.py
@@ -1 +1,12 @@
 import numpy
+
+numpy.set_printoptions(sign = ' ') # 1.13 compatibility
+
+n, m = [int(s) for s in input().split()]
+l = []
+for _ in range(n):
+    l.append([int(s) for s in input().split()])
+arr = numpy.array(l)
+print(numpy.mean(arr, axis = 1))
+print(numpy.var(arr, axis = 0))
+print("%.11f" % numpy.std(arr, axis = None))
-- 
2.23.0

