From 70068587566c445544bff846f4ae7e1bc09f868e Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Fri, 14 Feb 2020 19:08:02 +0000
Subject: [add] hackerrank, 'sherlock and array', javascript port

---
 .../sherlock.and.array.js                     | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/hackerrank/algorithms.-.search.-.easy.-.sherlock.and.array/sherlock.and.array.js b/hackerrank/algorithms.-.search.-.easy.-.sherlock.and.array/sherlock.and.array.js
index 9e16598..6c7bc53 100644
--- a/hackerrank/algorithms.-.search.-.easy.-.sherlock.and.array/sherlock.and.array.js
+++ b/hackerrank/algorithms.-.search.-.easy.-.sherlock.and.array/sherlock.and.array.js
@@ -24,8 +24,22 @@ function readLine() {
 
 // Complete the balancedSums function below.
 function balancedSums(arr) {
-
-
+    let l_a = arr.length;
+    if (l_a < 1)
+        return "NO";
+    let lsum = 0;
+    let rsum = arr.slice(1).reduce((a, x) => a + x, 0);
+    if (lsum === rsum)
+        return "YES";
+    let idx = 0;
+    while (idx < l_a - 1) {
+      lsum += arr[idx];
+      idx += 1;
+      rsum -= arr[idx];
+      if (lsum === rsum)
+          return "YES";
+    }
+    return "NO";
 }
 
 function main() {
@@ -45,4 +59,3 @@ function main() {
 
     ws.end();
 }
-
-- 
2.25.0

