From 1e4ff7774a2decfb20c2371fd811b8e3281c4210 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sat, 1 Feb 2020 18:49:10 +0000
Subject: [add] hackerrank, 'the love-letter mystery', go solution

---
 .../the.love.letter.mystery.go                        | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/hackerrank/algorithms.-.strings.-.easy.-.the.love.letter.mystery/the.love.letter.mystery.go b/hackerrank/algorithms.-.strings.-.easy.-.the.love.letter.mystery/the.love.letter.mystery.go
index b9ebe3b..b06ec18 100644
--- a/hackerrank/algorithms.-.strings.-.easy.-.the.love.letter.mystery/the.love.letter.mystery.go
+++ b/hackerrank/algorithms.-.strings.-.easy.-.the.love.letter.mystery/the.love.letter.mystery.go
@@ -12,7 +12,16 @@ import (
 
 // Complete the theLoveLetterMystery function below.
 func theLoveLetterMystery(s string) int32 {
-
+    ops := float64(0)
+    i := 0
+    i2 := len(s) - 1
+    chars := []rune(s)
+    for i < i2 {
+        ops += math.Abs(float64(chars[i2] - chars[i]))
+        i++
+        i2--
+    }
+    return int32(ops)
 }
 
 func main() {
-- 
2.24.1

