From 8f55b0c5ded0bcc6382b700b82d1335f9218fde1 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sat, 9 Jun 2018 19:13:15 +0100
Subject: [add] hackerrank, 'counting sort #2', javascript port

---
 .../counting.sort.2.js                                | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hackerrank/algorithms.-.sorting.-.easy.-.counting.sort.2/counting.sort.2.js b/hackerrank/algorithms.-.sorting.-.easy.-.counting.sort.2/counting.sort.2.js
index 235c6ca..d175f7b 100644
--- a/hackerrank/algorithms.-.sorting.-.easy.-.counting.sort.2/counting.sort.2.js
+++ b/hackerrank/algorithms.-.sorting.-.easy.-.counting.sort.2/counting.sort.2.js
@@ -26,6 +26,17 @@ function readLine() {
 
 // Complete the countingSort function below.
 function countingSort(arr) {
+    var n = arr.length;
+    var idxs = new Array(n).fill(0);
+    arr.forEach(x => {
+        idxs[x] += 1;
+    });
+    var res = [];
+    for (var x = 0; x < n; x++) {
+        if (idxs[x] != 0)
+            res = res.concat(new Array(idxs[x]).fill(x));
+    }
+    return res;
 }
 
 function main() {
-- 
2.23.0

