From 72745406d8376b97d80e428c5b64751f4efd74cf Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sun, 6 Oct 2019 09:59:21 +0100
Subject: [add] hackerrank, 'manasa and stones', c++ port

---
 .../manasa.and.stones.cpp                             | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.cpp b/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.cpp
index 8679db3..bd7c3bd 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.cpp
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.cpp
@@ -4,8 +4,15 @@ using namespace std;
 
 // Complete the stones function below.
 vector<int> stones(int n, int a, int b) {
-
-
+    /*
+    combinations of up to n-1 (account for 0 stone) a or b stones
+    calc x*a + ((n-1)-x)*b for 0 <= x < n - 1 ..order ascending
+    */
+    set<int> sums;
+    for (int x = 0; x < n; x++) sums.insert(x * a + ((n - 1) - x) * b);
+    vector<int> sums_;
+    copy(sums.begin(), sums.end(), back_inserter(sums_));
+    return sums_;
 }
 
 int main()
-- 
2.23.0

