From cce352b7f757cc69e60fa21c0c65f68fa1ef4bbe Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sat, 5 Oct 2019 09:10:56 +0100
Subject: [add] hackerrank, 'manasa and stones', python solution

---
 .../manasa.and.stones.py                                   | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.py b/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.py
index 508bb1f..ea5b7cd 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.py
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.manasa.and.stones/manasa.and.stones.py
@@ -6,8 +6,15 @@ import random
 import re
 import sys
 
+
 # Complete the stones function below.
 def stones(n, a, b):
+    """
+    combinations of up to n-1 (account for 0 stone) a or b stones
+    calc x*a + ((n-1)-x)*b for 0 <= x < n - 1 ..order ascending
+    """
+    return sorted([x * a + ((n - 1) - x) * b for x in range(0, n)])
+
 
 if __name__ == '__main__':
     fptr = open(os.environ['OUTPUT_PATH'], 'w')
-- 
2.23.0

