From 598e9e31f9f2568453338bee3f4fcffc8949334f Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Tue, 3 Dec 2019 23:40:38 +0000
Subject: [add] hackerrank, 'service lane', python solution

---
 .../service.lane.py                                          | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.service.lane/service.lane.py b/hackerrank/algorithms.-.implementation.-.easy.-.service.lane/service.lane.py
index aa6057a..7bc4c38 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.service.lane/service.lane.py
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.service.lane/service.lane.py
@@ -6,8 +6,12 @@ import random
 import re
 import sys
 
+
 # Complete the serviceLane function below.
 def serviceLane(n, widths, cases):
+    #print(f"widths {widths} | cases: {cases}")
+    return [min(widths[x[0]:x[1] + 1]) for x in cases] 
+
 
 if __name__ == '__main__':
     fptr = open(os.environ['OUTPUT_PATH'], 'w')
@@ -31,4 +35,3 @@ if __name__ == '__main__':
     fptr.write('\n')
 
     fptr.close()
-
-- 
2.23.0

