From 8b69ac57c10f8d618ba7493482420a81e38b56df Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Mon, 27 May 2019 11:23:31 +0100
Subject: [add] hackerrank, 'jumping on the clouds', javascript port

---
 .../jumping.on.the.clouds.js                          | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.jumping.on.the.clouds/jumping.on.the.clouds.js b/hackerrank/algorithms.-.implementation.-.easy.-.jumping.on.the.clouds/jumping.on.the.clouds.js
index f8cbf6b..56648d2 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.jumping.on.the.clouds/jumping.on.the.clouds.js
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.jumping.on.the.clouds/jumping.on.the.clouds.js
@@ -26,8 +26,15 @@ function readLine() {
 
 // Complete the jumpingOnClouds function below.
 function jumpingOnClouds(c) {
-
-
+    var idx_last = c.length - 1;
+    var jumps = 0;
+    var idx = 0;
+    while (idx < idx_last) {
+        idx = (idx + 2 <= idx_last && c[idx + 2] == 0) ? idx + 2 : idx + 1;
+        jumps += 1;
+        //console.log(`"idx: ${idx}, jumps: ${jumps}`);
+    }
+    return jumps;
 }
 
 function main() {
-- 
2.23.0

