From 691651f48840d42b94c59d6ffb970527efa5e313 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 27 Nov 2019 08:31:06 +0000
Subject: [add] hackerrank, 'fair rations', python solution

---
 .../fair.rations.py                           | 26 ++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.py b/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.py
index 832ed88..8b4faff 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.py
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.fair.rations/fair.rations.py
@@ -6,8 +6,33 @@ import random
 import re
 import sys
 
+
 # Complete the fairRations function below.
 def fairRations(B):
+    """
+    1 2 3
+    2 3 3
+    2 4 4
+
+    1 3 1
+    2 4 1
+
+    reject odd number of odds
+    """
+
+    l_odds = len([x for x in B if x % 2 == 1])
+    if l_odds % 2 == 1:
+        return "NO"
+
+    mod = 0
+    l_B = len(B)
+    for l in range(l_B - 1):
+        if B[l] % 2 == 1:
+            B[l] += 1
+            B[l + 1] += 1
+            mod += 2
+    return mod
+
 
 if __name__ == '__main__':
     fptr = open(os.environ['OUTPUT_PATH'], 'w')
@@ -21,4 +46,3 @@ if __name__ == '__main__':
     fptr.write(str(result) + '\n')
 
     fptr.close()
-
-- 
2.23.0

