From 30b3232abac01e946f3a87bd50ca4b9e5d3e057c Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sun, 3 Nov 2019 15:24:18 +0000
Subject: [add] hackerrank, 'halloween sale', c++ port

---
 .../halloween.sale.cpp                         | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.halloween.sale/halloween.sale.cpp b/hackerrank/algorithms.-.implementation.-.easy.-.halloween.sale/halloween.sale.cpp
index 1df87bd..e35a0f5 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.halloween.sale/halloween.sale.cpp
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.halloween.sale/halloween.sale.cpp
@@ -7,7 +7,23 @@ vector<string> split_string(string);
 // Complete the howManyGames function below.
 int howManyGames(int p, int d, int m, int s) {
     // Return the number of games you can buy
-
+    auto funds = s;
+    auto min = m;
+    auto last_book = p;
+    auto drop = d;
+    if (last_book > funds) {
+        return 0;
+    }
+    auto count = 0;
+    while (last_book > min && funds > last_book) {
+        funds -= last_book;
+        last_book -= drop;
+        count++;
+    }
+    if (last_book <= funds) {
+        count += funds / min;
+    }
+    return count;
 }
 
 int main()
-- 
2.23.0

