From a13ef4c1dfd337bd5169473bbfd80084f74bbadf Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 6 Jun 2018 18:26:25 +0100
Subject: [add] hackerrank, 'insertion sort - part 1', c# solution

---
 .../insertion.sort.part.1.cs                             | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hackerrank/algorithms.-.sorting.-.easy.-.insertion.sort.part.1/insertion.sort.part.1.cs b/hackerrank/algorithms.-.sorting.-.easy.-.insertion.sort.part.1/insertion.sort.part.1.cs
index 7053149..c23cb2c 100644
--- a/hackerrank/algorithms.-.sorting.-.easy.-.insertion.sort.part.1/insertion.sort.part.1.cs
+++ b/hackerrank/algorithms.-.sorting.-.easy.-.insertion.sort.part.1/insertion.sort.part.1.cs
@@ -16,6 +16,15 @@ class Solution {
 
     // Complete the insertionSort1 function below.
     static void insertionSort1(int n, int[] arr) {
+        var shift = arr[n - 1];
+        var idx = n - 1;
+        while (idx - 1 > 0 && arr[idx - 1] > shift) {
+            arr[idx] = arr[idx - 1];
+            Console.WriteLine(String.Join(" ", arr));
+            idx--;
+        }
+        arr[idx] = shift;
+        Console.WriteLine(String.Join(" ", arr));
     }
 
     static void Main(string[] args) {
-- 
2.23.0

