From 6e2772fd6d4b5b586b2358749b90fad451ebe3a9 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 6 Jun 2018 09:06:00 +0100
Subject: [add] hackerrank, 'designer pdf viewer', c# solution

---
 .../designer.pdf.viewer.cs                                     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.designer.pdf.viewer/designer.pdf.viewer.cs b/hackerrank/algorithms.-.implementation.-.easy.-.designer.pdf.viewer/designer.pdf.viewer.cs
index 29eded7..e14c67d 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.designer.pdf.viewer/designer.pdf.viewer.cs
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.designer.pdf.viewer/designer.pdf.viewer.cs
@@ -16,6 +16,9 @@ class Solution {
 
     // Complete the designerPdfViewer function below.
     static int designerPdfViewer(int[] h, string word) {
+        var width = word.Length;
+        var height = word.ToCharArray().Select(c => h[(int)c - 97]).OrderBy(x => x).Last();
+        return width * height;
     }
 
     static void Main(string[] args) {
-- 
2.23.0

