From a72cf5f7fe2dce00a1ae4812d593ba18d7efab64 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 4 Jul 2018 08:41:22 +0100
Subject: [add] hackerrank, 'tree inorder traversal', python solution

---
 .../tree.inorder.traversal.py                         | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hackerrank/data.structures.-.trees.-.easy.-.tree.inorder.traversal/tree.inorder.traversal.py b/hackerrank/data.structures.-.trees.-.easy.-.tree.inorder.traversal/tree.inorder.traversal.py
index ebec860..0c2492b 100644
--- a/hackerrank/data.structures.-.trees.-.easy.-.tree.inorder.traversal/tree.inorder.traversal.py
+++ b/hackerrank/data.structures.-.trees.-.easy.-.tree.inorder.traversal/tree.inorder.traversal.py
@@ -42,6 +42,17 @@ self.info (the value of the node)
 """
 def inOrder(root):
     #Write your code here
+    def dump(root):
+        values = [];
+        if root.left:
+            values += dump(root.left)
+        values.append(root.info)
+        if root.right:
+            values += dump(root.right)
+        return values
+
+    print(" ".join([str(x) for x in dump(root)]))
+
 
 tree = BinarySearchTree()
 t = int(input())
-- 
2.23.0

