From d37e59d159d7e1742044c26402f040b8d0caaf13 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Tue, 4 Jun 2019 08:47:38 +0100
Subject: [add] hackerrank, 'cut the sticks', c# port

---
 .../cut.the.sticks.cs                                 | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/hackerrank/algorithms.-.implementation.-.easu.-.cut.the.sticks/cut.the.sticks.cs b/hackerrank/algorithms.-.implementation.-.easu.-.cut.the.sticks/cut.the.sticks.cs
index 6f8e00e..82389d3 100644
--- a/hackerrank/algorithms.-.implementation.-.easu.-.cut.the.sticks/cut.the.sticks.cs
+++ b/hackerrank/algorithms.-.implementation.-.easu.-.cut.the.sticks/cut.the.sticks.cs
@@ -16,8 +16,15 @@ class Solution {
 
     // Complete the cutTheSticks function below.
     static int[] cutTheSticks(int[] arr) {
-
-
+        var arr2 = new List<int>(arr);
+        arr2.Sort();
+        var res = new List<int>();
+        while (arr2.Count() > 0) {
+            //Console.WriteLine($"[{arr2.Count()}] {String.Join(' ', arr2.AsEnumerable().Select(x => x.ToString()))}");
+            res.Add(arr2.Count());
+            arr2 = arr2.Where(x => x > arr2[0]).ToList();
+        }
+        return res.ToArray();
     }
 
     static void Main(string[] args) {
-- 
2.23.0

