From 6748cd829e13b910197e664b9cbb412d6155869f Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Tue, 5 Jun 2018 15:09:14 +0100
Subject: [add] hackerrank, 'picking numbers', c# solution

---
 .../picking.numbers.cs                                | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hackerrank/algorithms.-.implementation.-.easy.-.picking.numbers/picking.numbers.cs b/hackerrank/algorithms.-.implementation.-.easy.-.picking.numbers/picking.numbers.cs
index 5837e9b..d7b5dfb 100644
--- a/hackerrank/algorithms.-.implementation.-.easy.-.picking.numbers/picking.numbers.cs
+++ b/hackerrank/algorithms.-.implementation.-.easy.-.picking.numbers/picking.numbers.cs
@@ -16,6 +16,17 @@ class Solution {
 
     // Complete the pickingNumbers function below.
     static int pickingNumbers(int[] a) {
+        var res = 1;
+        var last_count = -1;
+        a.GroupBy(x => x).OrderBy(g => g.Key).ToList().ForEach(g => {
+            if (last_count > -1) {
+                // max concurrent pair
+                if (last_count + g.Count() > res)
+                    res = last_count + g.Count();
+            }
+            last_count = g.Count();
+        });
+        return res;
     }
 
     static void Main(string[] args) {
-- 
2.23.0

